homeassistant/components/remootio/config_flow.py

```diff
@@ -0,0 +1,179 @@
+                validation_result = await validate_input(self.hass, user_input)
+            except UnsupportedRemootioDeviceError:
+                _LOGGER.debug("Remootio device isn't supported", exc_info=True)
+                return self.async_abort(reason="unsupported_device")
+            except vol.MultipleInvalid as ex:
+                _LOGGER.error(
+                    "Invalid user input. MultipleInvalid.Errors [%s]", ex.errors
+                )
+                for error in ex.errors:
+                    _LOGGER.debug(
+                        "Error [%s] Path [%s]", error.__class__.__name__, error.path[0]
+                    )
+                    if isinstance(error, RequiredFieldInvalid):
+                        errors[str(error.path[0])] = f"{error.path[0]}_required"
+                    else:
+                        errors[str(error.path[0])] = f"{error.path[0]}_invalid"
+            except RemootioClientConnectionEstablishmentError:
+                _LOGGER.error("Can't connect to Remootio device")
+                errors["base"] = "cannot_connect"
+            except RemootioClientAuthenticationError:
+                _LOGGER.error("Can't authenticate by the Remootio device")
+                errors["base"] = "invalid_auth"
+            except Exception:  # pylint: disable=broad-except
+                _LOGGER.exception("Unexpected exception/error")
+                errors["base"] = "unknown"
+            else:
+                await self.async_set_unique_id(
+                    validation_result[CONF_DATA][CONF_SERIAL_NUMBER]
+                )
+                self._abort_if_unique_id_configured(validation_result[CONF_DATA])
+
+                return self.async_create_entry(
+                    title=validation_result[CONF_TITLE],
+                    data=validation_result[CONF_DATA],
+                )
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {
+                    vol.Optional(
+                        CONF_HOST,
+                        default=user_input.get(CONF_HOST, vol.UNDEFINED),
+                    ): vol.Coerce(str),
+                    vol.Optional(
+                        CONF_API_SECRET_KEY,
+                        default=user_input.get(CONF_API_SECRET_KEY, vol.UNDEFINED),
+                    ): vol.Coerce(str),
+                    vol.Optional(
+                        CONF_API_AUTH_KEY,
+                        default=user_input.get(CONF_API_AUTH_KEY, vol.UNDEFINED),
+                    ): vol.Coerce(str),
+                    vol.Optional(
+                        CONF_DEVICE_CLASS,
+                        default=user_input.get(
+                            CONF_DEVICE_CLASS, CoverDeviceClass.GARAGE
+                        ),
+                    ): vol.All(
+                        vol.Coerce(str),
+                        vol.In([CoverDeviceClass.GARAGE, CoverDeviceClass.GATE]),
+                    ),
```

Comment:
As mentioned in another comment, please remove the device class setting from here